extensions/s3/src/main/java/io/deephaven/extensions/s3/S3SeekableChannelProvider.java

```diff
@@ -15,16 +15,23 @@
 import java.nio.channels.SeekableByteChannel;
 import java.nio.file.Path;
 
+import static io.deephaven.extensions.s3.S3Instructions.MAX_FRAGMENT_SIZE;
+
 /**
  * {@link SeekableChannelsProvider} implementation that is used to fetch objects from AWS S3 instances.
  */
 final class S3SeekableChannelProvider implements SeekableChannelsProvider {
 
+    private static final BufferPool bufferPool = new SegmentedBufferPool(MAX_FRAGMENT_SIZE);
+
     private final S3AsyncClient s3AsyncClient;
     private final S3Instructions s3Instructions;
-    private final BufferPool bufferPool;
 
     S3SeekableChannelProvider(@NotNull final S3Instructions s3Instructions) {
+        if (s3Instructions.fragmentSize() > MAX_FRAGMENT_SIZE) {
+            throw new IllegalArgumentException("Fragment size " + s3Instructions.fragmentSize() + " is larger than " +
+                    " maximum allowed " + MAX_FRAGMENT_SIZE);
+        }
```

Comment:
This is redundant. `S3Instructions` cannot be built with an out of range size. Maybe you want to compare to a `POOLED_BUFFER_SIZE` (`= MAX_FRAGMENT_SIZE`) or something for future safety?